File: app/Data/Protocols/Db/MySql/Transaction/AddTransactionRepository.php
<?php

namespace App\Data\Protocols\Db\MySql\Transaction;

use App\Domain\Model\Deposit;
use App\Domain\Model\Transaction;
use App\Domain\Model\Withdraw;

interface AddTransactionRepository
{
    public function add (Deposit $deposit, Withdraw $withdraw) : Transaction;
}

File: src/main/kotlin/com/leonardovee/effectivesuccotash/data/usecase/AddTransactionRepository.kt
package com.leonardovee.effectivesuccotash.data.usecase

import com.leonardovee.effectivesuccotash.domain.model.Deposit
import com.leonardovee.effectivesuccotash.domain.model.Transaction
import com.leonardovee.effectivesuccotash.domain.model.Withdraw
import org.springframework.stereotype.Component

@Component
interface AddTransactionRepository {
    fun addTransaction(deposit: Deposit, withdraw: Withdraw): Transaction
}